### PR TITLE
Prometheus: Add POST support to client

### LINKED ISSUE #60242: Prometheus datasource POSTs to Backend via URL
**What happened**:
The Prometheus datasource doesn't use the request body if the POST method is selected

**What you expected to happen**:
Grafana should use the request body instead of the url

**How to reproduce it (as minimally and precisely as possible)**:
1. Setup Grafana with Prometheus
2. Execute a large query
3. Check HTTP logs on Prometheus

**Anything else we need to know?**:

**Environment**:
- Grafana version: v9.3.1
- Data source type & version: Prometheus
- OS Grafana is installed on: Debian
- User OS & Browser: Chrome
- Grafana plugins: default
- Others: -

### PR DESCRIPTION
**Pull request by fionera**
**What is this feature?**

The Prometheus Client rewrite is missing the correct handling of POST requests

**Why do we need this feature?**

Do larger queries on Prometheus

**Who is this feature for?**

-

**Which issue(s) does this PR fix?**:

Fixes #60242 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by gtk-grafana** [comment]
@fionera Thanks for the submission! It looks like some of the unit tests are failing on this branch, can you take a look and update those unit tests, and add some new ones to cover this functionality?

@toddtreece is this going to break existing code that expects the query to be coming in from the URL?

This is a good feature, however I'm thinking that we might want to let people configure this in their settings, to provide backward compatibility as there might be certain environments that have security/firewall rules set up for GET requests, and not POST, which could break if we switch over to POST, instead of allowing users the ability to switch between the two.

**Comment by fionera** [comment]
> This is a good feature, however I'm thinking that we might want to let people configure this in their settings, to provide backward compatibility as there might be certain environments that have security/firewall rules set up for GET requests, and not POST, which could break if we switch over to POST, instead of allowing users the ability to switch between the two.

This feature was already implemented in v8, so it doesn't need a feature gate. The Library used in v8 had always first tried POST and then had a fallback to GET. In the backend Grafana was using a middleware to always let the first request fail if the method was set to GET. This PR is basically reimplementing a feature that was overlooked when reimplementing the client.

As to the Unit-Test, I just pushed a fix for it :) Turns out in 1c503908 was this already noticed but just changed in the unit-test

**Comment by sakjur** [comment]
@fionera It's not your PR, GitHub is having some issues with fetching PRs using the `refs/pull/60243/head` format, we're trying to get in contact with GitHub about that 🙂 BuildKite are also seeing this and have opened a report about it on their status page: https://www.buildkitestatus.com/incidents/l4059rl7kdfs

**Comment by fionera** [comment]
Ah ok thank you :) Was already confused whats happening there. But yeah this PR is just porting previously existing functionality to the new client

**Comment by fionera** [comment]
Btw this is relatively important since because of the missing support, a lot of dashboards wont work because the Queries are too long for URLs

**Review by itsmylife — APPROVED** [review]
LGTM 🚀 
I just asked for additional reviews.

File: pkg/tsdb/prometheus/client/client_test.go [hunk 1]
```diff
@@ -80,4 +82,60 @@ func TestClient(t *testing.T) {
 			require.Equal(t, "http://localhost:9090/api/v1/series?match%5B%5D=ALERTS&start=1655272558&end=1655294158", doer.Req.URL.String())
 		})
 	})
+	t.Run("QueryRange", func(t *testing.T) {
```
Comment (marefr) [inline pkg/tsdb/prometheus/client/client_test.go:85]:
```suggestion

	t.Run("QueryRange", func(t *testing.T) {
```

File: pkg/tests/api/prometheus/prometheus_test.go [hunk 2]
```diff
@@ -134,7 +133,7 @@ func TestIntegrationPrometheus(t *testing.T) {
 
 		require.NotNil(t, outgoingRequest)
 		require.Equal(t, "/api/v1/query_range", outgoingRequest.URL.Path)
-		require.Contains(t, outgoingRequest.URL.String(), "&q1=1&q2=2")
+		require.Contains(t, outgoingRequest.URL.String(), "q1=1&q2=2")
```
Comment (marefr) [inline pkg/tests/api/prometheus/prometheus_test.go:136]: Maybe the following given Prometheus uses POST request?

```suggestion
		require.Contains(t, outgoingRequest.URL.String(), "?q1=1&q2=2")
```

**Review by marefr — APPROVED** [review]
LGTM, added some minor suggestions

**Review by marefr — APPROVED** [review]
LGTM

**Comment by LudoVio** [comment]
LGTM

**Comment by itsmylife** [comment]
@fionera Thanks for your contribution. 🚀

**Comment by fionera** [comment]
Awesome :rocket:

**Comment by yashsarma** [comment]
thanks for the patch. hey @itsmylife, looks like this wasn't included in the 9.3.2 release, do you know when the next release is going to be cut? 

Thanks

**Pull request merged**